src/postamble.js

```diff
@@ -225,7 +225,7 @@ function run(args) {
   if (preloadStartTime === null) preloadStartTime = Date.now();
 
   if (runDependencies > 0) {
-#if ASSERTIONS
+#if ASSERTIONS && !BINARYEN_ASYNC_COMPILATION
```

Comment:
Well, my thinking for the change above was that it's never an error when we do binaryen async compilation, but it is a sign that something is wrong otherwise. Don't you think `ASSERTIONS && !BINARYEN_ASYNC_COMPILATION` expresses that combination properly?